### DISCUSSION TITLE
Docker - shared data - which is the best approach?

### DISCUSSION THREAD

**Discussion by Stitch10925** [seaweedfs/seaweedfs#3921] [Q&A]
I'm currently playing around with Docker swarm and the idea is to use SeaweedFs to centralize data storage. I know there is a Docker driver for SeaweedFS, but it's not maintained anymore.

Other options would be to:
- Mount SeaweedFS on the Docker host and then bind the docker container to that mount
- Add SeaweedFS directly into Docker and mount from there

However, I am struggling with what option would be best, because in the first case there could be multiple containers writing to the same mount, so concurrency might be an issue there. In the second option each container has its own mount, but how will SeaweedFS handle concurrency when writing to the file-system? Is it the last write that wins? Is there going to be a conflict? Further: Would it make sense to create a filer for each Docker Service so that they all have their own place to write to?

Or is it not advisable to run SeaweedFS in combination with Docker at all?

**Comment by chrislusf** [comment]
Do they need to share files? Better for each writer to write to its own folder.

**Reply by Stitch10925** [reply]
Yes, some will need to share files since they will be load-balanced workers.

**Reply by chrislusf** [reply]
Mount directly would be better. There may be some unknown bugs with the extra VM layer.
